File: ios/Classes/Models/MeshEvent.swift
import Foundation

/// Type of mesh event
enum MeshEventType: Int {
    case meshStarted = 0
    case meshStopped = 1
    case peerDiscovered = 2
    case peerConnected = 3
    case peerDisconnected = 4
    case messageReceived = 5
    case error = 6
}

/// Represents an event in the mesh network
struct MeshEvent {
    let type: MeshEventType
    let message: String?
    let data: [String: Any]?

    init(type: MeshEventType, message: String? = nil, data: [String: Any]? = nil) {
        self.type = type
        self.message = message
        self.data = data
    }

    /// Convert to a dictionary for sending to Flutter
    func toMap() -> [String: Any?] {
        return [
            "type": type.rawValue,
            "message": message,
            "data": data
        ]
    }
}

File: android/src/main/kotlin/com/ble_mesh/models/MeshEvent.kt
package com.ble_mesh.models

/**
 * Type of mesh event
 */
enum class MeshEventType(val value: Int) {
    MESH_STARTED(0),
    MESH_STOPPED(1),
    PEER_DISCOVERED(2),
    PEER_CONNECTED(3),
    PEER_DISCONNECTED(4),
    MESSAGE_RECEIVED(5),
    ERROR(6);

    companion object {
        fun fromInt(value: Int) = values().firstOrNull { it.value == value } ?: ERROR
    }
}

/**
 * Represents an event in the mesh network
 */
data class MeshEvent(
    val type: MeshEventType,
    val message: String? = null,
    val data: Map<String, Any>? = null
) {
    /**
     * Convert to a map for sending to Flutter
     */
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "type" to type.value,
            "message" to message,
            "data" to data
        )
    }
}

